auth/src/main/java/org/aerogear/android/ags/auth/AbstractAuthenticator.java

```diff
@@ -40,7 +42,5 @@ public void logout(final Principal principal) {
      * Returns the authentication service configuration
      * @return the authentication service configuration
      */
-    protected AuthServiceConfig getConfig() {
-        return config;
-    }
+    public ServiceConfiguration getServiceConfig() { return this.serviceConfig; }
```

Comment:
Is ServiceConfiguration an immutable object?
If it is not, different threads could change the content of the `ServiceConfiguration` object, thus changing the internal state of the Authenticator without any control by the latter, making it crash prone.

Reply:
Not too familiar with any core stuff. @secondsun would you have an answer to this?

Reply:
ServiceConfiguration is meant to be immutable.  If it isn't then feel free to make it so.

Reply:
Have made this change in a separate PR. Don't want this one to bloat much more. https://github.com/aerogear/aerogear-android-sdk/pull/24/files